File: Team_Workspace/Fatma_Ahmed/cpp_tasks/task4/task4.cpp
// task4.

#include <iostream>
using namespace std;

enum LightColor {
  RED,
  YELLOW,
  GREEN,

};

struct TrafficLight {
    LightColor color;
    int timer_seconds;
};

void updateLight(TrafficLight &light)
{
    switch (light.color) {
    case LightColor::RED:
        light.color = LightColor::GREEN;
        light.timer_seconds = 30;
        break;



    case LightColor::GREEN:
        light.color = LightColor::YELLOW;
        light.timer_seconds = 5;
        break;

    case LightColor::YELLOW:
        light.color = LightColor::RED;
        light.timer_seconds = 60;
        break;

    }

}

string getStateString(TrafficLight& light) {
    switch (light.color) {
    case LightColor::RED:
        return "RED";
        break;
    case LightColor::YELLOW:
        return "YELLOW";
        break;
    case LightColor::GREEN:
        return "GREEN";
        break;

    }
}

int main()
{
    TrafficLight test1;
    test1.color = LightColor::RED;
    updateLight(test1);
    updateLight(test1);


    cout << "Final state: " << getStateString(test1);
}

File: Team_Workspace/Abdelrahman-Adel/Tasks/task_4.py
from enum import Enum
from dataclasses import dataclass

class LightColor(Enum):
    RED = 1
    YELLOW = 2
    GREEN = 3


@dataclass
class TrafficLight:
    color: LightColor
    timer_seconds: int


def update_light(light: TrafficLight):
    if light.color == LightColor.RED:
        light.color = LightColor.GREEN
        light.timer_seconds = 30
    elif light.color == LightColor.GREEN:
        light.color = LightColor.YELLOW
        light.timer_seconds = 5
    elif light.color == LightColor.YELLOW:
        light.color = LightColor.RED
        light.timer_seconds = 60

traffic_light = TrafficLight(color=LightColor.RED, timer_seconds=60)

update_light(traffic_light)  
update_light(traffic_light)  

print(f"Final State: {traffic_light.color.name}, Timer: {traffic_light.timer_seconds}s")
